core/src/main/java/org/arquillian/smart/testing/spi/JavaSPILoader.java

```diff
@@ -0,0 +1,64 @@
+package org.arquillian.smart.testing.spi;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.ServiceLoader;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class JavaSPILoader {
+
+    public JavaSPILoader() {
+    }
+
+    public <S> Iterable<S> all(Class<S> service) {
```

Comment:
My opinion - instead of single char `S` I would use the whole word that represents the meaning of this parameter: `SERVICE` or `SERVICETYPE`
Same for the following occurences